sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala

```diff
@@ -220,6 +220,33 @@ class DataFrameWriterV2Suite extends QueryTest with SharedSparkSession with Befo
       Seq(Row(1L, "a"), Row(2L, "b"), Row(4L, "d"), Row(5L, "e"), Row(6L, "f")))
   }
 
+  test("SPARK-36706 Overwrite: overwrite by expression: more than one filters") {
```

Comment:
should we test that the filters are changed? it seems we only check query answers but I think they should pass no matter with or without the fix?

Reply:
Yes, the test would pass without the fix. but couldn't figure out an easy way to get the `OverwriteByExpressionExec` node and check the filters on it.